src/cljc/rems/common/roles.cljc

```diff
@@ -32,7 +33,29 @@
             (boolean (some (set roles) context/*roles*)))
      :cljs (some (set roles) (:roles @(rf/subscribe [:identity])))))
 
+(defn is-with-organization?
+  "Determines if the user is part of an organization based on the passed in ID."
+  [organization-id]
+  #?(:cljs (let [owned-organizations @(rf/subscribe [:owned-organizations])]
+             (->> owned-organizations
+                  (map #(get-in % [:organization/id]))
+                  (in? organization-id)))))
+
+(defn has-permission?
+  "Does the current user have the proper permissions?
+   Permission is defined by having the proper role and is within an org."
+  [roles organization-id]
+  #?(:cljs
+     (and (apply has-roles? roles) (is-with-organization? organization-id))))
+
+#?(:cljs
+   (defn show-when
+     ([roles & body]
+      (clojure.core/when (apply has-roles? roles)
+        (into [:<>] body)))))
+
 #?(:cljs
-   (defn show-when [roles & body]
-     (clojure.core/when (apply has-roles? roles)
+   (defn show-when-correct-access
```

Comment:
If we do the above then this could be `show-when-owner-of` but perhaps that could be a regular `when` clause too. I'm not sure. Let's have a look after you have digested these ideas a bit?